### ISSUE TITLE
RLMObjectSchema  propertiesForClass crashed

### ISSUE THREAD

**Issue by iblacksun** [realm/realm-cocoa#5411]
## Expected Results
Work great

## Actual Results
Crash in RLMObjectSchema.mm line 224

## Steps to Reproduce

1. Launch app
2. Save objects

Will crash in line 224 due to array index out of bounds.

the ```prop``` is ```videos``` and ```index = 5```, but ```propArray.count = 4```
## Code Sample

```
public class ClassVideo: Object {
    public enum VideoType: Int {
        case trial = 1
        case authorization = 2
    }
    
    @objc public dynamic var ID = 0
    @objc public dynamic var name = ""
    @objc public dynamic var duration = 0
    @objc public dynamic var intro = ""
    public var videoType: VideoType = .authorization
    public let owner = LinkingObjects(fromType: ClassChapter.self, property: "videos")
}
```
Realm framework version: 3.0.0

Xcode version: 9.0

iOS/OSX version: 10.13

**Comment by iblacksun** [comment]
If i add ```videoType``` into ```ignoredProperties```, it's works greate.

**Comment by kishikawakatsumi** [comment]
@iblacksun You're correct. But I'm curious because `videoType` should be ignored automatically. (No need to add `videoType` into `ignoredProperties` explicitly) Could you please provide a reproducible sample project?

**Comment by StDogbert** [comment]
Same issue here. Same line throws. Different objects obviously. And I don't even have properties to ignore. 
Also lldb shows something strange:
(lldb) po propArray
<__NSArrayM 0x60c000845160>(

)

(lldb) po propArray.count
<nil>

**Comment by kishikawakatsumi** [comment]
@StDogbert Could you please share more details? We don't know your models and your code. Can you share your models and code that can reproduce the issue? Thanks.

**Comment by StDogbert** [comment]
@kishikawakatsumi I... I can't reproduce it anymore. I think I rebooted once since my original comment. Nothing else. Same project dependencies... I tested it a lot back then. I suppose its some black magic fuckery. Sorry for inconvenience.

**Comment by tgoyne** [comment]
Should be fixed by #5425.
